# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/DOMTypes.ipdlh
# Commit: 802512ee57ae
# Full Hash: 802512ee57aec9d975ab078c48b9f37956012bc6
# Author: Neil Deakin <neil@mozilla.com>
# Date: 2015-04-17 19:45:37
# Regressor Bug: 1071562
# File Overlap Count: 1
# Description:
#   Bug 1071562, e10s, support non-text types in clipboard (html, images, etc), r=smaug
# ==============================================================================

diff -r ffd1b16f058b -r 802512ee57ae dom/ipc/DOMTypes.ipdlh
--- a/dom/ipc/DOMTypes.ipdlh	Thu Apr 16 01:11:05 2015 +0300
+++ b/dom/ipc/DOMTypes.ipdlh	Thu Apr 16 15:38:12 2015 -0400
@@ -137,13 +137,24 @@
 
 union IPCDataTransferData
 {
-  nsString;
-  PBlob;
+  nsString;  // text
+  nsCString; // images
+  PBlob;     // files
+};
+
+struct IPCDataTransferImage
+{
+  uint32_t width;
+  uint32_t height;
+  uint32_t stride;
+  uint8_t format;
 };
 
 struct IPCDataTransferItem
 {
   nsCString flavor;
+  // The image details are only used when transferring images.
+  IPCDataTransferImage imageDetails;
   IPCDataTransferData data;
 };
 